_pytest/mark.py

```diff
@@ -4,17 +4,61 @@
 from operator import attrgetter
 from .compat import imap
 
+
 def alias(name):
     return property(attrgetter(name), doc='alias for ' + name)
 
 
+class ParameterSet(namedtuple('ParameterSet', 'values, marks, id')):
+    @classmethod
+    def param(cls, *value, **kw):
+        marks = kw.pop('marks', ())
+        
+        if isinstance(marks, MarkDecorator):
+            marks = marks,
+        else:
+            assert isinstance(marks, (tuple, list, set))
```

Comment:
Hmm I disagree... if I type `mark=` instead of `marks=`, why would I not want a glaring and loud error at that? Do you have an example of where just emitting a warning makes sense?